sprocs/grader_loads_current.sql

```diff
@@ -180,6 +182,18 @@ BEGIN
         desired_instances_by_history_users
     ));
 
+    -- ######################################################################
+    -- anti-flapping using historical information
+
+    SELECT coalesce(max(value), 0)
+    INTO desired_instances_history
+    FROM time_series
+    WHERE
+        name = 'desired_instances'
+        AND date > now() - history_interval;
```

Comment:
How big will this `history_interval` be in practice?  Something in the range of a few seconds?

Reply:
It's the existing value which is currently https://github.com/PrairieLearn/PrairieLearn/blob/19c59baa89e3786b7cc02fc7db3a1a26207d5398/lib/config.js#L84

This is too long, but at least it will reduce the flapping. We should change the grader autoscaler algorithm more, so this PR is just a stop-gap measure.

There are a number of other changes that I want to make to the grader hosts to bring them into line with workspace hosts. Things like clearer state tracking, better websocket communication back to the end user, better logging, etc. I need to file issues for all these things.